src/ansiblelint/rules/schema.py

```diff
@@ -75,6 +77,49 @@ class ValidateSchemaRule(AnsibleLintRule):
         "schema[vars]": "",
     }
 
+    def matchplay(self, file: Lintable, data: dict[str, Any]) -> list[MatchError]:
+        """Return matches found for a specific playbook."""
+        results: list[MatchError] = []
+
+        if not data or file.kind not in ("tasks", "handlers", "playbook"):
+            return results
+        # check at play level
+        become_method = data.get("become_method", None)
+        available_become_plugins = get_app().runtime.plugins.become
+        available_become_plugins_keys = available_become_plugins.keys()
+        if (
+            become_method
+            and not has_jinja(become_method)
+            and not available_become_plugins.get(become_method)
+        ):
+            results.append(
+                MatchError(
+                    message=f"'become_method' must be one of the currently installed plugins: {', '.join(available_become_plugins_keys)}",
+                    lintable=file or Lintable(""),
+                    rule=ValidateSchemaRule(),
+                    details=ValidateSchemaRule.description,
+                    tag=f"schema[{file.kind}]",
+                ),
+            )
+
+        # check at task level
+        tasks = data.get("tasks", [])
```

Comment:
This is not how you look inside tasks, we have `matchtask` method to implement which is called for each task within the rule. Look at parent implementation to see how is called.

Basically matchplay should never look inside `tasks` as we have a specific method that takes care of task level checks.